File: app/src/main/java/com/mumanit/foursquareclient/domain/GetVenuesInteractorImpl.java
package com.mumanit.foursquareclient.domain;

import com.mumanit.foursquareclient.domain.model.VenueData;
import com.mumanit.foursquareclient.data.VenuesDataManager;

import java.util.List;

import rx.Observable;

/**
 * Created by pmuciek on 7/8/17.
 */

public class GetVenuesInteractorImpl implements GetVenuesInteractor {

    private final VenuesDataManager venuesDataManager;

    public GetVenuesInteractorImpl(VenuesDataManager venuesDataManager) {
        this.venuesDataManager = venuesDataManager;
    }

    @Override
    public Observable<List<VenueData>> getVenues() {
        return venuesDataManager.getVenues();
    }
}

File: app/src/main/java/com/mumanit/foursquareclient/domain/GetVenuesInteractorImpl.kt
package com.mumanit.foursquareclient.domain

import com.mumanit.foursquareclient.data.VenuesDataManager
import com.mumanit.foursquareclient.domain.model.VenueData
import kotlinx.coroutines.flow.Flow

class GetVenuesInteractorImpl(private val venuesDataManager: VenuesDataManager) : GetVenuesInteractor {
    override fun getRecommendedVenuesWithFlow(): Flow<List<VenueData>> = venuesDataManager.getRecommendedVenuesWithFlow()
    override suspend fun getRecommendedVenues(): List<VenueData> = venuesDataManager.getRecommendedVenues()
}
